src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java

```diff
@@ -64,136 +55,65 @@ public CreateCaseCallbackService(
-
-        try {
-            String s2sToken = s2sTokenGenerator.generate();
-
-            SuccessfulTransformationResponse transformationResponse = transformationClient.transformExceptionRecord(
-                configItem.getTransformationUrl(),
-                exceptionRecord,
-                s2sToken
-            );
-
-            if (!ignoreWarnings && !transformationResponse.warnings.isEmpty()) {
-                // do not log warnings
-                return Validation.valid(new ProcessResult(
-                    transformationResponse.warnings,
-                    emptyList()
-                ));
-            }
-
-            log.info(
-                "Successfully transformed exception record for {} from exception record {}",
-                configItem.getService(),
-                caseIdAsString
-            );
-
-            long newCaseId = createNewCaseInCcd(
-                idamToken,
-                s2sToken,
-                userId,
-                exceptionRecord.poBoxJurisdiction,
-                transformationResponse.caseCreationDetails,
-                caseIdAsString
-            );
-
-            log.info(
-                "Successfully created new case for {} with case ID {} from exception record {}",
-                configItem.getService(),
-                newCaseId,
-                caseIdAsString
-            );
-
-            return Validation.valid(new ProcessResult(
-                ImmutableMap.of(CASE_REFERENCE, Long.toString(newCaseId))
-            ));
-        } catch (InvalidCaseDataException exception) {
-            if (BAD_REQUEST.equals(exception.getStatus())) {
-                throw exception;
-            } else {
-                return Validation.valid(new ProcessResult(
-                    exception.getResponse().warnings,
-                    exception.getResponse().errors
-                ));
-            }
-        } catch (Exception exception) {
-            log.error(
-                "Failed to create exception for service {} and exception record {}",
-                configItem.getService(),
-                caseIdAsString,
-                exception
-            );
-
-            return Validation.invalid(Array.of("Internal error. " + exception.getMessage()));
+        Validation<String, Void> event = isCreateNewCaseEvent(request.getEventId());
+        if (event.isInvalid()) {
+            return new ProcessResult(emptyList(), singletonList(event.getError()));
+        } else {
+            return hasServiceNameInCaseTypeId(request.getCaseDetails())
+                .map(serviceName -> {
+                    try {
+                        ServiceConfigItem serviceCfg = serviceConfigProvider.getConfig(serviceName);
+                        if (serviceCfg == null || serviceCfg.getTransformationUrl() == null) {
+                            return new ProcessResult(emptyList(), singletonList("Transformation URL is not configured"));
```

Comment:
I know that it's not about what this PR is for, but this error should be returned as a 500 response and the caseworker should get a generic error.